src/transformers/integrations.py

```diff
@@ -1127,15 +1129,33 @@ def _log_hyper_param_search_parameters(self, state):
         if state and hasattr(state, 'trial_params') and state.trial_params is not None:
             self._metadata_namespace['trial_params'] = state.trial_params
 
-    def _log_model_checkpoint(self, path: str):
-        self._metadata_namespace[self._target_checkpoints_namespace].upload_files(path)
+    def _log_model_checkpoint(self, source_directory: str, checkpoint: str):
+        target_path = relative_path = os.path.join(source_directory, checkpoint)
+
+        if self._volatile_checkpoints_dir is not None:
+            consistent_checkpoint_path = os.path.join(self._volatile_checkpoints_dir, checkpoint)
+            try:
+                shutil.copytree(relative_path, os.path.join(consistent_checkpoint_path, relative_path))
+                target_path = consistent_checkpoint_path
+            except IOError as e:
+                # TODO: NPT-12189 - Update warning copy
+                logger.warning(
+                    "NeptuneCallback was unable to made a copy of checkpoint due to I/O exception: '{}'."
+                    "Could fail trying to upload.".format(e)
+                )
+
+        self._metadata_namespace[self._target_checkpoints_namespace].upload_files(target_path)
 
         if self._should_clean_recently_uploaded_checkpoint and self._recent_checkpoint_path is not None:
             self._metadata_namespace[self._target_checkpoints_namespace].delete_files(self._recent_checkpoint_path)
 
-        self._recent_checkpoint_path = path
+        self._recent_checkpoint_path = relative_path
 
     def on_init_end(self, args, state, control, **kwargs):
+        self._volatile_checkpoints_dir = None
+        if self._log_checkpoints is not None and (args.overwrite_output_dir or args.save_total_limit is not None):
```

Comment:
Could you send me link to place when there's description of `transformers` behaviour and why only those two parameters (`args.overwrite_output_dir, args.save_total_limit):`) can lead to deleting data from disc?

Reply:
Sadly I just checked the `TrainerArguments` looking for anything that might end up with this behaviour and found this 2 arguments. Check https://huggingface.co/docs/transformers/v4.21.1/en/main_classes/trainer#transformers.TrainingArguments.save_total_limit for details.